aspnetcore/host-and-deploy/iis/troubleshoot.md

```diff
@@ -15,46 +15,140 @@ uid: host-and-deploy/iis/troubleshoot
+![Browser window showing the 502.5 Process Failure page](troubleshoot/_static/process-failure-page.png)
 
-When Kestrel starts normally behind IIS but the app won't run on the system after successfully running locally, an environment variable can be temporarily added to *web.config* to set the `ASPNETCORE_ENVIRONMENT` to `Development`. As long as the environment isn't overridden in app startup, setting the environment variable allows the [developer exception page](xref:fundamentals/error-handling) to appear when the app is run. Setting the environment variable for `ASPNETCORE_ENVIRONMENT` in this way is only recommended for staging/testing servers that aren't exposed to the Internet. Be sure to remove the environment variable from the *web.config* file when finished. For information on setting environment variables via *web.config*, see [environmentVariables child element of aspNetCore](xref:host-and-deploy/aspnet-core-module#setting-environment-variables).
+**500 Internal Server Error**  
+The app starts, but an error prevents the server from fulfilling the request.
 
-In most cases, enabling application logging assists in troubleshooting problems with the app or the reverse proxy. See [Logging](xref:fundamentals/logging/index) for more information.
+This error occurs within the app's code during startup or while creating a response. The response may contain no content, or the response may appear as a *500 Internal Server Error* in the browser. The Application Event Log usually states that the app started normally. From the server's perspective, that's correct. The app did start, but it can't generate a valid response. [Run the app at a command prompt](#run-the-app-at-a-command-prompt) on the server or [enable the ASP.NET Core Module stdout log](#aspnet-core-module-stdout-log) to troubleshoot the problem.
 
-The last troubleshooting tip pertains to apps that fail to run after upgrading either the .NET Core SDK on the development machine or package versions within the app. In some cases, incoherent packages may break an app when performing major upgrades. Most of these issues can be fixed by:
+## Default startup limits
 
-* Deleting the `bin` and `obj` folders in the project.
-* Clearing package caches at `%UserProfile%\.nuget\packages\` and `%LocalAppData%\Nuget\v3-cache`.
-* Restoring and rebuilding the project.
-* Confirming that the prior deployment on the server has been completely deleted prior to re-deploying the app.
+The ASP.NET Core Module is configured with a default *startupTimeLimit* of 120 seconds. When left at the default value, an app may take up to two minutes to start before the module logs a process failure. For information on configuring the module, see [Attributes of the aspNetCore element](xref:host-and-deploy/aspnet-core-module#attributes-of-the-aspnetcore-element).
+
+## Troubleshoot app startup errors
+
+### Application Event Log
```

Comment:
Many users encounter the Process Error while still in development (IIS Express). The easiest way to inspect these is to start the app under the debugger. We should lead with that.

Reply:
... but the topic is focused on IIS-hosted troubleshooting. Wouldn't that be better (if it isn't already there) in the Error Handling topic, which should concentrate on errors during development?

Reply:
Fair, so long as they can get directed to the correct doc.

Reply:
@Tratcher The error handling topic says this ...

> Hosting can only show an error page for a captured startup error if the error occurs after host address/port binding. If any binding fails for any reason, the hosting layer logs a critical exception, the dotnet process crashes, and no error page is displayed.

... but it sounds like ur saying that the last bit of that isn't true for IIS/Express. The process crashes and IIS/Express provides a 502.5 in the browser. What's there today applies just to Kestrel (apparently).

Reply:
Right, those Hosting comments apply to the application process. ANCM adds an error page in front of that.

Reply:
@guardrex  can you do this?
>Many users encounter the Process Error while still in development (IIS Express). The easiest way to inspect these is to start the app under the debugger. We should lead with that.
It doesn't matter if it's duplicate info. Just a short sentence and like to another topic.

Reply:
I did, but it was a bit buried under the *Learn to debug using Visual Studio* link. I've moved it right to the intro.

Wherever we put it, let's please *not* put it down where @Tratcher wrote the comment. That's a section for troubleshooting steps.

An alternative spot is in the *App startup errors* section under the **502.5 Process Failure** heading.